### ISSUE TITLE
[YSQL] Dangling sequences are created when CREATE TABLE has conficts.

### ISSUE THREAD

**Issue by nocaway** [yugabyte/yugabyte-db#2602]
This issue is reported by @ddorian .

In the following example, two sequences were created including a dangling-sequence "tab_id_seq1" that is not associated with any table.

```
postgres=# CREATE TABLE tab(id SERIAL PRIMARY KEY);
CREATE TABLE
postgres=# CREATE TABLE tab(id SERIAL PRIMARY KEY);
ERROR:  relation "tab" already exists
postgres=# \d
             List of relations
 Schema |    Name     |   Type   |  Owner   
--------+-------------+----------+----------
 public | tab         | table    | postgres
 public | tab_id_seq  | sequence | postgres
 public | tab_id_seq1 | sequence | postgres
postgres=# DROP TABLE tab;
DROP TABLE
postgres=# \d
             List of relations
 Schema |    Name     |   Type   |  Owner   
--------+-------------+----------+----------
 public | tab_id_seq1 | sequence | postgres
```

**Comment by m-iancu** [comment]
We should confirm if this can still happen today -- I think that might have been fixed by the transactional DDL work: b68b84bfd46b6a771d828c66477a477aa0f71844.
If the example above works now, we can just close the issue.

**Comment by ddorian** [comment]
It's fixed. Tested on 2.3.2:

```
postgres=# CREATE TABLE tab(id SERIAL PRIMARY KEY);
CREATE TABLE
postgres=# \d
             List of relations
 Schema |    Name    |   Type   |  Owner   
--------+------------+----------+----------
 public | tab        | table    | yugabyte
 public | tab_id_seq | sequence | yugabyte
(2 rows)

postgres=# CREATE TABLE tab(id SERIAL PRIMARY KEY);
ERROR:  relation "tab" already exists
postgres=# \d
             List of relations
 Schema |    Name    |   Type   |  Owner   
--------+------------+----------+----------
 public | tab        | table    | yugabyte
 public | tab_id_seq | sequence | yugabyte
(2 rows)

postgres=# DROP TABLE tab;
DROP TABLE
postgres=# \d
Did not find any relations.
```
